packages/legend-application-repl/src/stores/dataCube/grid/DataCubeGridState.ts

```diff
@@ -139,4 +148,166 @@ export class DataCubeGridState extends DataCubeQuerySnapshotSubscriber {
       await this.dataCube.replStore.client.getGridClientLicenseKey(),
     );
   }
+
+  generateCSVFile = () => {
+    const params = {
+      fileName: 'csv_data.csv',
+    };
+    if (this._client) {
+      this._client.exportDataAsCsv(params);
+    } else {
+      console.error('Grid API not set');
+    }
+  };
+
+  generateExcelFile = () => {
+    if (!this._client) {
+      throw new Error('Grid API is not initialized.');
+    }
+    const params = {
+      fileName: 'excel_data.xlsx',
+      processCellCallbacks: (cell: any) => {
+        return cell ? cell.toString() : '';
+      },
+    };
+    this._client.exportDataAsExcel(params);
+  };
+
+  blobToBase64 = (blob: Blob): Promise<string> => {
+    return new Promise((resolve, reject) => {
+      const reader = new FileReader();
+      reader.onloadend = () => {
+        const result = reader.result;
+        if (typeof result === 'string') {
+          const base64Data = result.split(',')[1];
+          if (base64Data) {
+            resolve(base64Data);
+          } else {
+            console.error('Base64 conversion failed: Invalid base64 data.');
```

Comment:
we avoid using console.error, let's consolidate the error and log it out using `logger.error`, later, also help me put a TODO cuz later we might add alert and use that to make these error more forward to users

```ts
// TODO: alert when failure/error occurs
```